app/src/main/java/com/ataulm/support/ColorAttributeResolvers.kt

```diff
@@ -0,0 +1,36 @@
+package com.ataulm.support
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.content.res.Resources
+import android.util.TypedValue
+import androidx.annotation.AttrRes
+import androidx.annotation.ColorInt
+import androidx.appcompat.content.res.AppCompatResources
+
+/**
+ * If it's possible to use a [ColorStateList] instead of a color int,
+ * then use [Context.resolveColorStateListAttribute] instead.
+ */
+@ColorInt
+fun Context.resolveColorAttribute(@AttrRes attr: Int): Int {
+    return resolveColorStateListAttribute(attr).defaultColor
+}
+
+fun Context.resolveColorStateListAttribute(@AttrRes attr: Int): ColorStateList {
+    return TypedValue().let { value ->
+        theme.resolveAttribute(attr, value, true)
+        when {
+            value.type >= TypedValue.TYPE_FIRST_COLOR_INT && value.type <= TypedValue.TYPE_LAST_COLOR_INT -> {
+                // This is a color hex value. Use value.data
+                ColorStateList.valueOf(value.data)
```

Comment:
this is a bit of a waste but it's because there's only a `color` resource type for both plain colors and CSLs. 

It means we can be less strict with users on which function they use (they pass a theme attribute that's a plain color or CSL for either). This sounds like a bad thing, but the caller shouldn't know what the value of the theme attribute is, so they shouldn't be penalized for choosing the wrong one.

Instead, they should choose `resolveColorAttribute` vs `resolveColorStateListAttribute` depending on the property they're setting. If it supports both (e.g. TextView text color), use the CSL one, else use the plain one.